pyaedt/modules/Boundary.py

```diff
@@ -4519,3 +4520,232 @@ def _create_boundary(bound):
             raise Exception
     except Exception:  # pragma: no cover
         return None
+
+
+class BoundaryDictionary:
+    """
+    Handles Icepak transient and temperature-dependent boundary condition assignments.
+
+    Parameters
+    ----------
+    assignment_type : str
+        Type of assignment represented by the class. Options are `"Temp Dep"``
+        and ``"Transient"``.
+    function_type : str
+        Variation function to assign. If ``assignment_type=="Temp Dep"``,
+        the function can only be ``"Piecewise Linear"``. Otherwise, the function can be
+        ``"Exponential"``, ``"Linear"``, ``"Piecewise Linear"``, ``"Power Law"``,
+        ``"Sinusoidal"``, and ``"Square Wave"``.
+    """
+
+    def __init__(self, assignment_type, function_type):
+        if assignment_type not in ["Temp Dep", "Transient"]:  # pragma : no cover
+            raise AttributeError("The argument {} for ``assignment_type`` is not valid.".format(assignment_type))
+        if assignment_type == "Temp Dep" and function_type != "Piecewise Linear":  # pragma : no cover
+            raise AttributeError(
+                "Temperature dependent assignments only support"
+                ' ``"Piecewise Linear"`` as ``function_type`` argument.'.format(assignment_type)
+            )
+        self.assignment_type = assignment_type
+        self.function_type = function_type
+
+    @property
+    def props(self):
+        return {
+            "Type": self.assignment_type,
+            "Function": self.function_type,
+            "Values": self._parse_value(),
+        }
+
+    @abstractmethod
```

Comment:
@lorenzovecchietti Why do we need this?

Reply:
Is it mandatory?

Reply:
No, I used it so if new child classes are added in the future the implementation of `_parse_value` will be mandatory. It is a good practice as  `_parse_value` is used in `props` of the base class.